File: Week2_arrays/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>
#include <math.h>

int main(void){
    string all_text = get_string("Text: ");
    int length = strlen(all_text);
    int sentence_count = 0;
    int letter_count = 0;
    int word_count = 1;
    for(int i=0;i<length;++i){
        if(all_text[i] == '.' ||all_text[i] == '!' || all_text[i] == '?'){
            sentence_count++;
        }
        if(isalpha(all_text[i])){
            letter_count++;
        }
        if(all_text[i]== ' '){
            word_count++;
        }
    }
    double L = ((double) letter_count/word_count)*100;
    double S = ((double) sentence_count/word_count)*100;

    double grade = 0.0588 * L - 0.296 * S - 15.8;
    double result = round(grade);
    int result1 = (int)result;
    if(result1<1){
        printf("Before Grade 1\n");
    }
    else if(result1>=16){
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %d\n",result1);
    }
    return 0;
}

File: Week6_python/sentimental-readability/readability.py
def get_string(prompt):
    return str(input(prompt))

all_text = get_string("Text: ")
lenght = len(all_text)
sentence_count = 0
letter_count = 0
word_count = 1
list_text = list(all_text)
for i in range (0, lenght):
    if list_text[i] == '.'  or list_text[i] == '!' or list_text[i] == '?' :
            sentence_count += 1
    if list_text[i].isalpha() == True:
         letter_count += 1
    if list_text[i] == ' ':
         word_count += 1

L = float(letter_count/word_count) * 100
S = float(sentence_count/word_count) * 100

grade = 0.0588 * L - 0.296 * S - 15.8
result = round(grade)
if result<1:
    print("Before Grade 1")
elif result>=16:
    print("Grade 16+")
else:
    print(f"Grade {result}");

